File: joke.py
import requests


def get_joke(category, flags):
    url = f'https://v2.jokeapi.dev/joke/{category}'
    params = {'blacklistFlags': flags}
    response = requests.get(url, params=params)
    if response.status_code == 200:
        joke_data = response.json()
        if 'type' in joke_data:
            if joke_data['type'] == 'single':
                return joke_data['joke']
            else:
                return f"{joke_data['setup']} - {joke_data['delivery']}"
        else:
            return "No joke found."

# print(get_joke('Programming', 'nsfw,religious'))

def search_jokes(term):
    url = 'https://v2.jokeapi.dev/joke/Any'
    params = {'contains': term}
    response = requests.get(url, params=params)
    if response.status_code == 200:
        joke_data = response.json()
        if 'type' in joke_data:
            if joke_data['type'] == 'single':
                return joke_data['joke']
            else:
                return f"{joke_data['setup']} - {joke_data['delivery']}"
        else:
            return "No joke found."

# print(search_jokes('bar'))

File: joke.js
function getJoke() {
    var category = document.getElementById('category').value;
    fetch('http://localhost:5000/display_joke/' + category)
        .then(response => response.json())
        .then(data => {
            document.getElementById('joke').innerText = data.joke;
        });
    return false;
}

function searchJoke(event) {
    event.preventDefault();
    var term = document.getElementById('term').value;
    fetch('http://localhost:5000/search_joke/' + term)
        .then(response => response.json())
        .then(data => {
            document.getElementById('search-joke').innerText = data.joke;
        });
}